### PR TITLE
Release preparation: Run codespell.

### PR DESCRIPTION
**Pull request by drwells**
I ran codespell, our own detect double-word typos script, and typos.

### CODE REVIEW

**Review by marcfehling — APPROVED** [review]

**Review by bangerth — APPROVED** [review]
Very nice!

How hard would you think it is to hook whatever script you ran into the CI?

**Comment by drwells** [comment]
These spellcheckers are not a good fit for the CI since we use a lot of non-English words. Some examples:
1. Both `typos` and `codespell` want to correct `compute_2nd_derivatives` in different ways
2. Many of our abbreviations (e.g., `Numer.`) are correct, but they don't know this
3. We use a lot of short-hand variables (like `eu`) that are also technically correct
4. non-English family names (Linz, Stoer)

We'd have to maintain a pretty substantial additional word list to make these tools run cleanly. I think its wiser to just run them at release time and manually inspect the output.

**Comment by bangerth** [comment]
Ah, thanks!

File: contrib/python-bindings/source/export_triangulation.cc [hunk 1]
```diff
@@ -229,7 +229,7 @@ namespace python
     "with respect to the origin. Additional points are sums of these dim    \n"
     "vectors. Colorizing is done according to hyper_rectangle().            \n"
     "Note: This function silently reorders the vertices on the cells to     \n"
-    "lexicographic ordering (see GridReordering::reoder_grid()). In other   \n"
+    "lexicographic ordering (see GridReordering::reorder_grid()). In other  \n"
```
[the remark this answers is not in the document]
Comment (drwells) [reply contrib/python-bindings/source/export_triangulation.cc]: Nice catch. Fixed!

**Review by masterleinad — APPROVED** [review]

File: include/deal.II/lac/solver_gmres.h [hunk 2]
```diff
@@ -944,7 +944,7 @@ namespace internal
                 is_dealii_compatible_distributed_vector<VectorType>::value,
                 VectorType> * = nullptr>
     double
-    substract_and_norm(
+    subtract_and_norm(
```
Comment (kronbichler) [inline include/deal.II/lac/solver_gmres.h:947]: :smile:

**Review by kronbichler — APPROVED** [review]
Thank you. It is always interesting to see how many typos are slipping through, and it is great to have this step once per year.

**Pull request merged**